DPL: replace LOG with Signposts

ktf opened the pull request:
DPL: replace LOG with Signposts

Move a bunch of log messages to use signposts for:

* AsyncQueue
* TimesliceIndex
* CommonServices
* DataRelayer

Also modified the async queue to get the oldest possible timeslice as index, so
that we can use that to track executions.

alibuild commented:
Error while checking build/O2/fullCI for a7e9e2d34f37dd9da749277710c26731fe41a0d5 at 2024-03-28 08:29:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
task timeout reached .. killing all processes


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/b4b684d4353557849067b056ea4e5ca252c22b91/slc8_x86-64/o2checkcode/1.0-local294/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12947/a7e9e2d34f37dd9da749277710c26731fe41a0d5/build_O2_fullCI/pretty.html).

ktf committed:
DPL: replace LOG with Signposts

Move a bunch of log messages to use signposts for:

* AsyncQueue
* TimesliceIndex
* CommonServices
* DataRelayer

Also modified the async queue to get the oldest possible timeslice as index, so
that we can use that to track executions.
